plexapi/library.py

```diff
@@ -549,16 +580,25 @@ def search(self, title=None, sort=None, maxresults=999999, libtype=None, **kwarg
             args['sort'] = self._cleanSearchSort(sort)
         if libtype is not None:
             args['type'] = utils.searchType(libtype)
-        # iterate over the results
-        results, subresults = [], '_init'
+
+        results = []
+        subresults = []
         args['X-Plex-Container-Start'] = 0
-        args['X-Plex-Container-Size'] = min(X_PLEX_CONTAINER_SIZE, maxresults)
-        while subresults and maxresults > len(results):
+        args['X-Plex-Container-Size'] = container_size
```

Comment:
Doesn't `maxresults` still apply? If `maxresults` is smaller than `container_size` (or implicitly `X_PLEX_CONTAINER_SIZE`) then you are currently retrieving too many items.

Reply:
I have never used the maxresults argument, so i dropped the ball on this. I don't like the maxresults default value so I'm gonna change this.

Reply:
Yeah I don't like that default value either.